packages/fl_query/lib/src/core/query.dart

```diff
@@ -184,6 +184,8 @@ class Query<DataType, ErrorType>
           if (error is ErrorType) _errorController.add(error);
```

Comment:
maybe we'll have to throw the error here? otherwise this future will always be completed and we won't be able to handle the error.

```
try {
  await mutation.mutate()
} catch (e) {
  // treat error
}
```